### PR TITLE
Wire up the Activity Resource to the OTLP exporter

### PR DESCRIPTION
**Pull request by pcwiese**
This change injects the real Resource from the Activity instead of the empty one. I verified that Spans sent to the collector have the Resource values set.

### CODE REVIEW

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs [hunk 1]
```diff
@@ -146,8 +143,7 @@ internal static OtlpTrace.Span ToOtlpSpan(this Activity activity)
                     continue;
                 }
 
-                // TODO: Retrieve resources for trace (not library/ActivitySource)
-                var resource = fakeResource;
+                var resource = activity.GetResource() ?? Resource.Empty;
```
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs:146]: Empty is returned by GetResource is null, so do you need the null check here?

Comment (pcwiese) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs:146]: no, we strictly don't. I can remove that bit

**Comment by cijothomas** [comment]
@pcwiese Thanks! I see that you closed/reopened this PR - was this done to re-trigger CI? If you see any unstable test, please paste it here, so we can investigate and fix them.

**Comment by pcwiese** [comment]
> @pcwiese Thanks! I see that you closed/reopened this PR - was this done to re-trigger CI? If you see any unstable test, please paste it here, so we can investigate and fix them.

Correct, I did it to re-run the ci pipeline. Here is the failing test;

```
  X OpenTelemetry.Trace.Export.Test.BatchingActivityProcessorTests.ExporterIsSlowerThanDelay [2s 65ms]
  Error Message:
   Expected at least 20, got 16
Expected: True
Actual:   False
  Stack Trace:
     at OpenTelemetry.Trace.Export.Test.BatchingActivityProcessorTests.WaitForActivities(TestActivityExporter exporter, Int32 activityCount, TimeSpan timeout) in D:\a\opentelemetry-dotnet\opentelemetry-dotnet\test\OpenTelemetry.Tests\Implementation\Trace\Export\BatchingActivityProcessorTests.cs:line 451
   at OpenTelemetry.Trace.Export.Test.BatchingActivityProcessorTests.ExporterIsSlowerThanDelay() in D:\a\opentelemetry-dotnet\opentelemetry-dotnet\test\OpenTelemetry.Tests\Implementation\Trace\Export\BatchingActivityProcessorTests.cs:line 130
```

I really think people should have the ability to re-run a pipeline without having to resort to weirdness like closing/re-opening, pushing innocuous changes up, etc. Of course,  deterministic unit tests are better ;)

**Comment by cijothomas** [comment]
> > @pcwiese Thanks! I see that you closed/reopened this PR - was this done to re-trigger CI? If you see any unstable test, please paste it here, so we can investigate and fix them.
> 
> Correct, I did it to re-run the ci pipeline. Here is the failing test;
> 
> ```
>   X OpenTelemetry.Trace.Export.Test.BatchingActivityProcessorTests.ExporterIsSlowerThanDelay [2s 65ms]
>   Error Message:
>    Expected at least 20, got 16
> Expected: True
> Actual:   False
>   Stack Trace:
>      at OpenTelemetry.Trace.Export.Test.BatchingActivityProcessorTests.WaitForActivities(TestActivityExporter exporter, Int32 activityCount, TimeSpan timeout) in D:\a\opentelemetry-dotnet\opentelemetry-dotnet\test\OpenTelemetry.Tests\Implementation\Trace\Export\BatchingActivityProcessorTests.cs:line 451
>    at OpenTelemetry.Trace.Export.Test.BatchingActivityProcessorTests.ExporterIsSlowerThanDelay() in D:\a\opentelemetry-dotnet\opentelemetry-dotnet\test\OpenTelemetry.Tests\Implementation\Trace\Export\BatchingActivityProcessorTests.cs:line 130
> ```
> 
> I really think people should have the ability to re-run a pipeline without having to resort to weirdness like closing/re-opening, pushing innocuous changes up, etc. Of course, deterministic unit tests are better ;)

Thanks. Some instability in tests got introduced in the last week in BatchingProcessorTests. Hunting them down one by one :)

We'll surely enable better way to rerun CI. It would be as simple as commenting `gh action run all` or something like that. Will get to it after some beta releases are done.

**Comment by cijothomas** [comment]
@TimothyMothra Do you know the magic we need to trigger CI?  `/azure pipeline run all` was for azure pipelines, but this is GH Actions.

**Pull request merged**